### DISCUSSION TITLE
Where can I find all the attributes of leafmap.Map?

### DISCUSSION THREAD

**Discussion by tikitong** [opengeos/leafmap#148] [Q&A] [answered]
Hello, 

I have a small question about the attributes of the leafmap.Map class. In [this notebook](https://leafmap.org/notebooks/01_leafmap_intro/) we find attributes like `draw_control`, `measure_control`, `fullscreen_control`, `attribution_control`. There is also `draw_export` which is very useful. But I don't see what class they inherit from. It says [here](https://leafmap.org/leafmap/#leafmap.leafmap.Map) that it's ipyleaflet.Map but in the [ipyleaflet doc](https://ipyleaflet.readthedocs.io/en/latest/api_reference/map.html) I think they don't appear, right? 

If so, where are they from? And where can I find the other attributes?  Thank you very much.

**Comment by giswqs** [comment, marked as answer]
`leafmap.Map` inherits all `ipyleaflet.Map` attributes with some additional ones only available for leafmap. They are not documented in the docs, but you can find them within the `__init()__` function.

https://github.com/giswqs/leafmap/blob/master/leafmap/leafmap.py#L19

**Reply by tikitong** [reply]
@giswqs Indeed thank you very much for your answer, but I still don't see the `draw_export` attribute in the constructor (__init__), Is it normal?

**Reply by giswqs** [reply]
`draw_export` is under the foliumap module. See https://github.com/giswqs/leafmap/blob/master/leafmap/foliumap.py#L45

**Reply by tikitong** [reply]
@giswqs I see it is a mix between ipyleaflet.Map,  added attributes **and folium.Map**. Maybe it should be added in the doc. Thanks again for your answer and even more for leafmap!

**Reply by giswqs** [reply]
Not all `ipyleaflet.Map` attributes are available for `folium.Map`. The goal of leafmap is to make a unified API so that users can the same code to create different maps by simply switching the plotting backend, i.e., `import leafmap`  vs `import leafmap.foliumap as leafmap`. 

https://leafmap.org/get-started/#important-note
https://leafmap.org/notebooks/00_key_features/#set-control-visibility
